### PR TITLE
Ratelimiter Acquire options

### PR DESCRIPTION
**Pull request by asmyasnikov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: internal/ratelimiter/ratelimiter.go [hunk 1]
```diff
@@ -196,22 +151,61 @@ func (c *client) AcquireResource(
...
+			}
+			_, _ = c.service.AcquireResource(
+				ctx,
+				&request,
+			)
+		}()
+	}
+
+	switch {
+	case errors.IsOpError(err, errors.StatusTimeout, errors.StatusTimeout):
```
[the remark this answers is not in the document]
Comment (asmyasnikov) [reply internal/ratelimiter/ratelimiter.go]: ack

File: internal/ratelimiter/ratelimiter.go [hunk 2]
```diff
@@ -196,22 +151,61 @@ func (c *client) AcquireResource(
...
+	default:
+		request.Units = &Ydb_RateLimiter.AcquireResourceRequest_Used{
+			Used: amount,
 		}
 	}
-	_, err = c.service.AcquireResource(c.ctx(ctx), &request)
-	return
+
+	switch acquireOptions.Type() {
+	case options.AcquireTypeAcquire:
```
[the remark this answers is not in the document]
Comment (asmyasnikov) [reply internal/ratelimiter/ratelimiter.go]: ack

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: internal/ratelimiter/ratelimiter.go [hunk 3]
```diff
@@ -196,22 +152,64 @@ func (c *client) AcquireResource(
...
 ) (err error) {
-	var request Ydb_RateLimiter.AcquireResourceRequest
-	if isUsedAmount {
-		request = Ydb_RateLimiter.AcquireResourceRequest{
+	var (
+		acquireOptions = options.NewAcquire(opts...)
+		request        = Ydb_RateLimiter.AcquireResourceRequest{
 			CoordinationNodePath: coordinationNodePath,
 			ResourcePath:         resourcePath,
 			Units:                &Ydb_RateLimiter.AcquireResourceRequest_Used{Used: amount},
```
[the remark this answers is not in the document]
Comment (asmyasnikov) [reply internal/ratelimiter/ratelimiter.go]: ack

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,8 +2,8 @@
 * Fixed closing proxy clients
 * Added `ydb.Connection.With(opts ...ydb.CustomOption)` for taking proxy `ydb.Connection` with some redefined options
 * Added `ydb.MetaRequestType` and `ydb.MetaTraceID` aliases to internal `meta` package constants
-* Added `ydb.WithRatelimiterOptions()` option
-* Added `ydb.WithCustomCredentials()` `CustomOption`
+* Added `ydb.WithCustomCredentials()` option
+* Refactored `ydb.Ratelimiter().AcquireResource()` method (added options for define type of acquire request)
```
[the remark this answers is not in the document]
Comment (asmyasnikov) [reply CHANGELOG.md]: ack

File: internal/ratelimiter/options/acquire.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	return h.timeout
+}
+
+func (h *acquireOptionsHolder) Type() AcquireType {
+	return h.acquireType
+}
+
+type AcquireOption func(h *acquireOptionsHolder)
+
+func WithTimeout(timeout time.Duration) AcquireOption {
```
[the remark this answers is not in the document]
Comment (asmyasnikov) [reply internal/ratelimiter/options/acquire.go]: discard

**Pull request merged**